airflow/providers/jenkins/sensors/jenkins.py

```diff
@@ -42,21 +43,28 @@ def __init__(
         jenkins_connection_id: str,
         job_name: str,
         build_number: int | None = None,
+        target_states: Iterable[str],
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.job_name = job_name
         self.build_number = build_number
         self.jenkins_connection_id = jenkins_connection_id
+        self.target_states = target_states or ["SUCCESS", "FAILED"]
 
     def poke(self, context: Context) -> bool:
         self.log.info("Poking jenkins job %s", self.job_name)
         hook = JenkinsHook(self.jenkins_connection_id)
-        is_building = hook.get_build_building_state(self.job_name, self.build_number)
+        build_number = self.build_number or hook.get_latest_build_number(self.job_name)
+        is_building = hook.get_build_building_state(self.job_name, build_number)
 
         if is_building:
             self.log.info("Build still ongoing!")
             return False
-        else:
-            self.log.info("Build is finished.")
+
+        build_result = hook.get_build_result(self.job_name, build_number)
+        self.log.info(f"Build is finished, result is {build_result}.")
+        if build_result in self.target_states:
             return True
+        else:
+            raise AirflowException("The build result does not meet the target states.")
```

Comment:
Can this be more specific? Can the user do something to fix this, or is this an impossible situation the user should tell the Jenkins admin?